src/utils/scripts.ts

```diff
@@ -37,3 +39,87 @@ export const parseP2PKH = (buff: Buffer, network: Network): P2PKH => {
 
   return new P2PKH(helpers.encodeAddress(addressHash, network), { timelock });
 };
+
+/**
+* Parse Data output script
+*
+* @param {Buffer} buff Output script
+*
+* @return {ScriptData} ScriptData object
+*/
+export const parseScriptData = (buff: Buffer): ScriptData => {
+  // We should clone the buffer being sent in order to never mutate
+  // what comes from outside the library
+  let scriptBuf = _.clone(buff);
+  if (scriptBuf.length < 2) {
+    // At least 1 byte for len data and 1 byte for OP_CHECKSIG
+    throw new ParseScriptError('Invalid output script.');
+  }
+
+  // The expected len will be at least 2 bytes
+  // 1 for the script len and 1 for the OP_CHECKSIG in the end
+  let expectedLen = 2;
+  let dataBytesLen;
+
+  // If we have OP_PUSHDATA1 as first byte, the second byte has the length of data
+  // otherwise, the first byte already has the length of data
+  if (scriptBuf[0] === OP_PUSHDATA1[0]) {
+    expectedLen += 1;
+    dataBytesLen = scriptBuf[1];
+  } else {
+    dataBytesLen = scriptBuf[0];
+  }
+
+  // Set the expected length
+  expectedLen += dataBytesLen;
+
+  if (expectedLen !== scriptBuf.length) {
+    // The script has different qty of bytes than expected
+    throw new ParseScriptError('Invalid output script.');
+  }
+
+  if (scriptBuf[expectedLen - 1] !== OP_CHECKSIG[0]) {
+    // Last byte must be an OP_CHECKSIG
+    throw new ParseScriptError('Invalid output script.');
```

Comment:
Should the error message better describe what is happening in each case?
Like: `Invalid output script. Last byte must be an OP_CHECKSIG`.

Reply:
Makes sense. Added more details in all error messages here https://github.com/HathorNetwork/hathor-wallet-lib/pull/295/commits/8d6c46cfdf5d96b377f4a33118805f191095380f